project_mirror.go

```diff
@@ -0,0 +1,162 @@
+//
+// Copyright 2017, Sander van Harmelen
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"time"
+)
+
+// ProjectMirrorService handles communication with the project mirror
+// related methods of the GitLab API.
+type ProjectMirrorService struct {
+	client *Client
+}
+
+// ProjectMirror represents a project mirror configuration.
+//
+// GitLAb API docs:
+// https://docs.gitlab.com/ce/api/remote_mirrors.html
+type ProjectMirror struct {
+	Enabled                bool        `json:"enabled"`
+	ID                     int         `json:"id"`
+	LastError              interface{} `json:"last_error"`
+	LastSuccessfulUpdateAt time.Time   `json:"last_successful_update_at"`
+	LastUpdateAt           time.Time   `json:"last_update_at"`
+	LastUpdateStartedAt    time.Time   `json:"last_update_started_at"`
```

Comment:
Are all these times always returned by the API?

Reply:
Documentation doesn't state it, but testing myself looks like these properties are always returned.